### PR TITLE
ref(hc): OrganizationIntegration moved to replication superclass

### PR DESCRIPTION
**Pull request by corps**
Just moving this model over to consolidate on this new approach.

### CODE REVIEW

**Review by markstory — COMMENTED** [review]
Looking good. Nice to see the outbox superclasses working out in more scenarios.

File: src/sentry/models/apikey.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ class ApiKeyStatus:
 class ApiKey(ReplicatedControlModel, HasApiScopes):
     __relocation_scope__ = RelocationScope.Global
     category = OutboxCategory.API_KEY_UPDATE
-    replication_version = 2
+    replication_version = 3
```
Comment (corps) [inline src/sentry/models/apikey.py:31]: Intended to bump this in a previous PR.  Doing so will make it possible for me to trigger a backfill via options.

**Review by dashed — APPROVED** [review]

File: tests/sentry/integrations/jira_server/test_integration.py [hunk 2]
```diff
@@ -1042,7 +1044,8 @@ def test_migrate_plugin(self):
             key=f"{self.plugin.slug}:tid", group_id=group2.id, value="BAR-1"
         )
         org_integration = OrganizationIntegration.objects.get(integration_id=self.integration.id)
-        org_integration.config.update({"issues_ignored_fields": ["reporter", "test"]})
+        with unguarded_write(router.db_for_write(OrganizationIntegration)):
```
Comment (GabeVillalobos) [inline tests/sentry/integrations/jira_server/test_integration.py:1047]: A bit curious, but why does this need an unguarded write?

**Review by GabeVillalobos — APPROVED** [review]
Looks good!

File: tests/sentry/integrations/jira_server/test_integration.py (hunk 2 above)
Comment (corps) [reply tests/sentry/integrations/jira_server/test_integration.py:1026]: It did not, so I removed. I think I was aggressive heuristically.

**Pull request merged**